apps/admin_app/lib/admin_app_web/controllers/product_controller.ex

```diff
@@ -100,6 +101,22 @@ defmodule AdminAppWeb.ProductController do
     end
   end
 
+  def update_inventory_tracking(conn, %{"product" => product_params} = params) do
+    with %ProductSchema{} = product <- ProductModel.get(params["product_id"]) do
+      tracking_level = product_params["inventory_tracking"]
+
+      case tracking_level do
+        "product" ->
+          Inventory.set_inventory_tracking(product, tracking_level, params["stock"])
+
+        _ ->
+          Inventory.set_inventory_tracking(product, tracking_level)
+      end
+
```

Comment:
how about using pattern matching in the `set_inventory_tracking` function directly.

```elixir
with %ProductSchema{} = product <- ProductModel.get(params["product_id"])  do
  Inventory.set_inventory_tracking(product, tracking_level, params["stock"])
end
```

```elixir
def set_inventory_tracking(product, inventory_tracking, %{"inventory_tracking" => "product", "stock" => stock_params}) do
....
end

def set_inventory_tracking(product, inventory_tracking, %{ "stock" => stock_params}) do
...
end
```

Reply:
done